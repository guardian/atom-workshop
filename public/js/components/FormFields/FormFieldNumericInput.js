import React, {PropTypes} from 'react';
import ShowErrors from '../Utilities/ShowErrors';
import { errorPropType } from '../../constants/errorPropType';

export default class FormFieldNumericInput extends React.Component {

  static propTypes = {
    fieldLabel: PropTypes.string,
    fieldName: PropTypes.string,
    fieldValue: PropTypes.number,
    fieldPlaceholder: PropTypes.string,
    fieldErrors: PropTypes.arrayOf(errorPropType),
    onUpdateField: PropTypes.func
  };

  onUpdate = (e) => {
    const value = e.target.value;
    const parsedValue = parseFloat(value);

    if (parsedValue) {
      this.props.onUpdateField(parsedValue);
    } else if (value === "") {
      this.props.onUpdateField(undefined);
    }
  }

  render() {
    return (
        <div>
          <label htmlFor={this.props.fieldName} className="form__label">{this.props.fieldLabel}</label>
          <input type="number" className="form__field" id={this.props.fieldName} placeholder={this.props.fieldPlaceholder || ''} value={this.props.fieldValue || ''} onChange={this.onUpdate}/>
          <ShowErrors errors={this.props.fieldErrors}/>
        </div>

    );
  }
}
