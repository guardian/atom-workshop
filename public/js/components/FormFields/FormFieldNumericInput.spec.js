import React from 'react';
import FormFieldNumericInput from './FormFieldNumericInput';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';

let fieldLabel = 'test',
    fieldName = 'test',
    fieldValue= 1;

test('Should render', () => {

  const updateFn = jest.fn();
  const component = renderer.create(
    <FormFieldNumericInput fieldLabel={fieldLabel} fieldName={fieldName} fieldValue={fieldValue} onUpdateField={updateFn} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Should call update function on change', () => {

  const updateFn = jest.fn();
  const input = shallow(
    <FormFieldNumericInput fieldLabel={fieldLabel} fieldName={fieldName} fieldValue={fieldValue} onUpdateField={updateFn} />
  );

  input.find('input').simulate('change', {target: {value: "test"}});
  expect(updateFn).toHaveBeenCalledTimes(0);
  input.find('input').simulate('change', {target: {value: "1"}});
  expect(updateFn).toHaveBeenCalledTimes(1);


});
