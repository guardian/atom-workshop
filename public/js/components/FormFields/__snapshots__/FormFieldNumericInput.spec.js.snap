exports[`test Should render 1`] = `
<div
  className="form__row">
  <label
    className="form__label"
    htmlFor="test">
    test
  </label>
  <input
    className="form__field"
    id="test"
    onChange={[Function]}
    placeholder=""
    type="number"
    value={1} />
</div>
`;
